resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -55,6 +55,34 @@ public TimeLimiterImpl(String name, TimeLimiterConfig timeLimiterConfig) {
         };
     }
 
+    @Override
+    public <T, F extends CompletionStage<T>> Supplier<CompletionStage<T>> decorateCompletionStage(Supplier<F> supplier) {
+        return () -> CompletableFuture.supplyAsync(() -> {
+            CompletableFuture<T> future = supplier.get().toCompletableFuture();
+            try {
+                return future.get(getTimeLimiterConfig().getTimeoutDuration().toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
I think we should avoid to do a blocking call.

Reply:
Please have a look at the delayer in this example.
http://iteratrlearning.com/java9/2016/09/13/java9-timeouts-completablefutures.html
What do you think?

Reply:
`executor.setRemoveOnCancelPolicy(true);` can be useful
 https://medium.com/@nilesh_mahajan/java-concurrency-pitfalls-of-scheduledexecutorservice-d7298b66e59f